core/auth/index.ts

```diff
@@ -0,0 +1 @@
+export { User } from './user'
```

Comment:
we need to export all the other interfaces and types from user.
plus this must be exported with type.
Please add isolatedModules to tsConfig